packages/tenant-consumer/src/model/converter.ts

```diff
@@ -0,0 +1,149 @@
+  kind: fromTenantMailKindV1(input.kind),
+});
+
+export const fromTenantFeatureV1 = (
+  input: TenantFeatureV1
+): TenantFeatureCertifier =>
+  match<TenantFeatureV1["sealedValue"], TenantFeatureCertifier>(
+    input.sealedValue
+  )
+    .with({ oneofKind: "certifier" }, ({ certifier }) => ({
+      type: "Certifier",
+      certifierId: certifier.certifierId,
+    }))
+    .with({ oneofKind: undefined }, () => {
+      throw new Error("Unable to deserialize TenantFeature");
+    })
+    .exhaustive();
+
+export const fromTenantVerifierV1 = (
+  input: TenantVerifierV1
+): TenantVerifier => ({
+  id: input.id,
+  verificationDate: new Date(Number(input.verificationDate)),
+  expirationDate: input.expirationDate
+    ? new Date(Number(input.expirationDate))
+    : undefined,
+  extensionDate: input.extensionDate
+    ? new Date(Number(input.extensionDate))
+    : undefined,
+});
+
+export const fromTenantRevokerV1 = (input: TenantRevokerV1): TenantRevoker => ({
+  id: input.id,
+  expirationDate: input.expirationDate
+    ? new Date(Number(input.expirationDate))
+    : undefined,
+  extensionDate: input.extensionDate
+    ? new Date(Number(input.extensionDate))
+    : undefined,
+  revocationDate: new Date(Number(input.revocationDate)),
+  verificationDate: new Date(Number(input.verificationDate)),
+});
+
+export const fromTenantAttributesV1 = (
+  input: TenantAttributeV1
+): TenantAttribute =>
+  match<TenantAttributeV1["sealedValue"], TenantAttribute>(input.sealedValue)
+    .with({ oneofKind: "certifiedAttribute" }, ({ certifiedAttribute }) => ({
+      id: certifiedAttribute.id,
+      assignmentTimestamp: new Date(
+        Number(certifiedAttribute.assignmentTimestamp)
+      ),
+      type: "certified",
+    }))
+    .with({ oneofKind: "verifiedAttribute" }, ({ verifiedAttribute }) => ({
+      id: verifiedAttribute.id,
+      assignmentTimestamp: new Date(
+        Number(verifiedAttribute.assignmentTimestamp)
+      ),
+      verifiedBy: verifiedAttribute.verifiedBy.map(fromTenantVerifierV1),
+      revokedBy: verifiedAttribute.revokedBy.map(fromTenantRevokerV1),
+      type: "verified",
+    }))
+    .with({ oneofKind: "declaredAttribute" }, ({ declaredAttribute }) => ({
+      id: declaredAttribute.id,
+      assignmentTimestamp: new Date(
+        Number(declaredAttribute.assignmentTimestamp)
+      ),
+      type: "declared",
+    }))
+    .otherwise(() => {
+      throw new Error("Booom"); // Ported "as is" from Scala codebase :D
+    });
+
+export const fromTenantV1 = (input: TenantV1): Tenant => {
+  /**
+   * The `externalId` field is required in the TenantV1 protobuf model but
+   * for some reasons the @protobuf-ts/protoc library generates it as optional.
+   * This issue has been reported here: https://github.com/timostamm/protobuf-ts/issues/340
+   */
+  if (!input.externalId) {
+    throw new Error(
+      `Error while deserializing TenantV1 (${input.id}): missing externalId`
+    );
+  }
```

Comment:
for future improvements I suggest to considerer a zod validation usages to handle parsing like this, for sure this issue is common in all protobuf deserialization.